apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/environment/Env.java

```diff
@@ -18,213 +18,233 @@
+      com.ctrip.framework.apollo.core.enums.Env.LOCAL.name());
+  public static final Env DEV = addEnvironment(
+      com.ctrip.framework.apollo.core.enums.Env.DEV.name());
+  public static final Env FWS = addEnvironment(
+      com.ctrip.framework.apollo.core.enums.Env.FWS.name());
+  public static final Env FAT = addEnvironment(
+      com.ctrip.framework.apollo.core.enums.Env.FAT.name());
+  public static final Env UAT = addEnvironment(
+      com.ctrip.framework.apollo.core.enums.Env.UAT.name());
+  public static final Env LPT = addEnvironment(
+      com.ctrip.framework.apollo.core.enums.Env.LPT.name());
+  public static final Env PRO = addEnvironment(
+      com.ctrip.framework.apollo.core.enums.Env.PRO.name());
+  public static final Env TOOLS = addEnvironment(
+      com.ctrip.framework.apollo.core.enums.Env.TOOLS.name());
+  public static final Env UNKNOWN = addEnvironment(
+      com.ctrip.framework.apollo.core.enums.Env.UNKNOWN.name());
+  // name of environment, cannot be null
+  private final String name;
+
+  /**
+   * Cannot create by other
+   *
+   * @param name
+   */
+  private Env(String name) {
+    this.name = name;
+  }
+
+  /**
+   * add some change to environment name trim and to upper
+   *
+   * @param envName
+   * @return
+   */
+  private static String getWellFormName(String envName) {
+    if (Strings.isBlank(envName)) {
+      return "";
     }
-
-    /**
-     * logic same as
-     * @see com.ctrip.framework.apollo.core.enums.EnvUtils transformEnv
-     * @param envName
-     * @return
-     */
-    public static Env transformEnv(String envName) {
-        if(Env.exists(envName)) {
-            return Env.valueOf(envName);
-        }
-        if (StringUtils.isBlank(envName)) {
-            return Env.UNKNOWN;
-        }
-        switch (envName.trim().toUpperCase()) {
-            case "LPT":
-                return Env.LPT;
-            case "FAT":
-            case "FWS":
-                return Env.FAT;
-            case "UAT":
-                return Env.UAT;
-            case "PRO":
-            case "PROD": //just in case
-                return Env.PRO;
-            case "DEV":
-                return Env.DEV;
-            case "LOCAL":
-                return Env.LOCAL;
-            case "TOOLS":
-                return Env.TOOLS;
-            default:
-                return Env.UNKNOWN;
-        }
-    }
-    
-    /**
-     * a environment name exist or not
-     * @param name
-     * @return
-     */
-    public static boolean exists(String name) {
-        name = getWellFormName(name);
-        return STRING_ENV_MAP.containsKey(name);
+    return envName.trim().toUpperCase();
+  }
+
+  /**
+   * logic same as
+   *
+   * @param envName the name we want to transform
+   * @return the env object matching the <code>envName</code>
+   * @see com.ctrip.framework.apollo.core.enums.EnvUtils transformEnv
+   */
+  public static Env transformEnv(String envName) {
```

Comment:
How about change to

```java
  public static Env transformEnv(String envName) {
    final String envWellFormName = getWellFormName(envName);
    // special case for production in case of typo
    if ("PROD".equalsIgnoreCase(envWellFormName)) {
      return Env.PRO;
    }

    // special case that FAT & FWS should map to FAT
    if ("FWS".equalsIgnoreCase(envWellFormName)) {
      return Env.FAT;
    }

    if (Env.exists(envWellFormName)) {
      return Env.valueOf(envWellFormName);
    }
    // cannot be found or blank name
    return Env.UNKNOWN;
  }
```